src/feedcontainer.cpp

```diff
@@ -109,6 +109,45 @@ void FeedContainer::sort_feeds(const FeedSortStrategy& sort_strategy)
 			}
 		});
 		break;
+	case FeedSortMethod::LAST_UPDATED_UNREAD:
+		std::stable_sort(
+			feeds.begin(), feeds.end(),
+		[&](std::shared_ptr<RssFeed> a, std::shared_ptr<RssFeed> b) {
+			if (a->unread_item_count() == 0 || b->unread_item_count() == 0) {
+				bool result = a->unread_item_count() > b->unread_item_count();
+				if (sort_strategy.sd == SortDirection::ASC) {
+					result = !result;
+				}
+				return result;
+			}
+			std::vector<std::shared_ptr<RssItem>> a_unread;
+			std::copy_if(a->items().begin(), a->items().end(),
+				std::back_inserter(a_unread),
+			[](const std::shared_ptr<RssItem>& item) {
+				return item->unread();
+			});
+			std::vector<std::shared_ptr<RssItem>> b_unread;
+			std::copy_if(b->items().begin(), b->items().end(),
```

Comment:
So this makes a filtered copy. I'm thinking that this is kind of inefficient, but also that efficiency does not matter so much here. From what I understand, newsboat is quite lazy in re-sorting. For example, there is no re-sorting when number of unread changes (considering unreadarticlescount sorting). Aside from restarting newsboat, it seems re-sorting can be provoked by changing which sort method is used (`g` key).

Reply:
From my own local testing, it indeed does not feel slow at all.
However, my list of feeds, and especially my list of unread feeds, is relatively small.
In general, I prefer to write things in an efficient way, but I would be fine with merging this as-is.

I did some experimenting locally with an alternative approach.
I might share it a bit later, still want to test it a bit more extensively.